alf/experience_replayers/experience_replay.py

```diff
@@ -120,10 +124,19 @@ def replay(self, sample_batch_size, mini_batch_length):
         raise NotImplementedError()  # Only supports replaying all!
 
     def replay_all(self):
-        return self._experience
+        assert self._experience, "No experience is observed yet!"
+        # batch major (B, T, ...)
+        exps = alf.nest.map_structure_up_to(
+            self._exp_sample, lambda e: torch.transpose(
+                torch.stack(e, dim=0), 0, 1), self._experience)
+        self.clear(
```

Comment:
```whole_replay_buffer_training = True``` does not necessarily need to clear all the buffer.
We have another flag ```clear_replay_buffer``` for explicitly specifying this. See:

https://github.com/HorizonRobotics/alf/blob/38c714928ab3e2f6a6c016dd9f7d71eba0ebf43a/alf/algorithms/off_policy_algorithm.py#L130-L133

Reply:
The name of this replayer is "one-time". So I believe that we should enforce this clear after replay.

Reply:
Yes, but isn't it true that  ```one-time-replayer``` is equivalent to replayer with ```whole_replay_buffer_training = True``` and ```clear_replay_buffer = True```?

Reply:
I was thinking that "one-time" basically overwrites these two values just in case people set them in a wrong way. But but default these two flags are indeed right for "one-time".

Reply:
I suggest we only keep whole_replay_buffer_training and clear_replay_buffer. Make sure their combination is valid. And choose appropriate replayer based on its value to make the config easy.

Reply:
updated